controllers/users.js

```diff
@@ -0,0 +1,65 @@
+const model = require ('../models/users')
+
+module.exports.getAll = function getAll(req, res){
+  model.find({}).exec((err, data) => {
+    if (err) return res.status(404).send()
+    res.json({ data })
+  })
+}
+
+module.exports.newUser = function newUser(req, res) {
+  let alreadyTaken = false 
+  const newUser = new model(req.body)
+  model.find({userName:newUser.userName}).exec((error,data) => {
+    if (error) return console.error(error)
+    for (var i = data.length - 1; i >= 0; i--) {
```

Comment:
you if data.length > 0 means that you are currently having duplicates, you do not neet to loop again because you are filtering in line 13